tools/codegen/gen.py

```diff
@@ -334,10 +337,39 @@ def go(f: NativeFunction) -> Optional[str]:
 
         assert target is Target.DEFINITION
 
-        def generate_defn(sig: CppSignature) -> str:
+        def generate_defn(sig: CppSignature, faithful: bool) -> str:
```

Comment:
`faithful` should be a property of `CppSignature`, no?

Edit: now that I've read the logic below, this seems even more straightforwardly true. `generate_defn` should be a simple consumer of a `CppSignature`. Any structuring of the signature itself should have already happened by the time we get here.

Reply:
Yes, agreed here. @smessmer, I think we did talk about this in chat, but I think I did not completely understand the problem you were posing me. I think the relevant comment from you was:

> So if I go with your approach and group the arguments into regular and out arguments, then the CppSignature doesn't have that info anymore but now needs to store an additional flag of if the out arguments are supposed to come before or after the regular arguments

And so to solve this problem, I proposed "where you continue to maintain a list, but then you have fingers into particular pieces of the rep. So CppSignatures will be a list of CppArgument, but then it also has an out_argument list which redundantly specifies which of these arguments were the out arguments so you can conveniently reference them in translation".

So you did the marking. But you failed to maintained the correctly ordered list of arguments in CppSignature so that you don't need the flag here anymore! (And it seems to me like maintaining direct pointers to out arguments, rather than flagging them on the arguments themselves, is easier, because then you don't need to do the partitioning business here.)